save-common/src/commonMain/kotlin/org/cqfn/save/core/reporter/Reporter.kt

```diff
@@ -0,0 +1,115 @@
+package org.cqfn.save.core.reporter
+
+import org.cqfn.save.core.config.ReportType
+import org.cqfn.save.core.plugin.Plugin
+import org.cqfn.save.core.plugin.PluginException
+import org.cqfn.save.core.result.TestResult
+
+import okio.BufferedSink
+
+/**
+ * A base interface for all reporters, that are capable of formatting results of SAVE execution
+ */
+interface Reporter {
+    /**
+     * A type of this reporter
+     */
+    val type: ReportType
```

Comment:
for config plugins you didn't want to have type

Reply:
Well, for reporters it's natural to have an explicit configuration, and not in toml, but rather in properties or even CLI. So I think explicitly listing types of reports is more appropriate, than hard-coding types of plugins